tests/std/tests/P0898R3_concepts/test.cpp

```diff
@@ -1487,68 +1487,83 @@ namespace test_constructible_from {
 namespace test_default_initializable {
     using std::default_initializable, std::initializer_list;
 
-#pragma warning(push)
-#pragma warning(disable : 4180) // qualifier applied to function type has no meaning; ignored
-    template <class T>
-    constexpr bool test() {
-        constexpr bool result = default_initializable<T>;
-        STATIC_ASSERT(default_initializable<T const> == result);
-        STATIC_ASSERT(default_initializable<T volatile> == result);
-        STATIC_ASSERT(default_initializable<T const volatile> == result);
-        return result;
-    }
-#pragma warning(pop)
-
-    STATIC_ASSERT(test<int>());
-    STATIC_ASSERT(test<int const>());
-    STATIC_ASSERT(test<double>());
-    STATIC_ASSERT(!test<void>());
-
-    STATIC_ASSERT(test<int*>());
-    STATIC_ASSERT(test<int const*>());
-
-    STATIC_ASSERT(test<int[2]>());
-    STATIC_ASSERT(test<char[3]>());
-    STATIC_ASSERT(test<char[5][3]>());
-    STATIC_ASSERT(!test<int[]>());
-    STATIC_ASSERT(!test<char[]>());
-    STATIC_ASSERT(!test<char[][3]>());
-
-    STATIC_ASSERT(!test<int&>());
-    STATIC_ASSERT(!test<int const&>());
-    STATIC_ASSERT(!test<int&&>());
-
-    STATIC_ASSERT(!test<int()>());
-    STATIC_ASSERT(!test<int (&)()>());
-
-    STATIC_ASSERT(!test<void()>());
-    STATIC_ASSERT(!test<void() const>());
-    STATIC_ASSERT(!test<void() volatile>());
-    STATIC_ASSERT(!test<void() &>());
-    STATIC_ASSERT(!test<void() &&>());
-
-    STATIC_ASSERT(test<EmptyClass>());
-    STATIC_ASSERT(test<EmptyUnion>());
-
-    STATIC_ASSERT(test<std::initializer_list<int>>());
-
-    STATIC_ASSERT(test<NotEmpty>());
-    STATIC_ASSERT(test<BitZero>());
-
-    STATIC_ASSERT(test<ExplicitDefault>());
-    STATIC_ASSERT(test<ExplicitMoveAbomination>());
-    STATIC_ASSERT(test<ExplicitCopyAbomination>());
-
-    STATIC_ASSERT(!test<NotDefaultConstructible>());
-    STATIC_ASSERT(!test<DeletedDefault>());
-
-    STATIC_ASSERT(!test<PurePublicDestructor>());
-    STATIC_ASSERT(!test<NotDefaultConstructible>());
+    STATIC_ASSERT(default_initializable<int>);
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, VSO-1084668
+    STATIC_ASSERT(!default_initializable<int const>);
+#else // ^^^ non-MSVC / MSVC vvv

```

Comment:
I'd slightly prefer to see "no workaround / workaround" so we know which side to keep.